typescript/t.d.ts

```diff
@@ -166,17 +179,21 @@ type ParseTReturn<
   : // Process plurals only if count is provided inside options
   TOpt['count'] extends number
   ? TOpt['ordinal'] extends boolean
-    ? ParseTReturnPluralOrdinal<Res, Key>
-    : ParseTReturnPlural<Res, Key>
+    ? ParseTReturnJoinArrays<ParseTReturnPluralOrdinal<Res, Key>, TOpt>
+    : ParseTReturnJoinArrays<ParseTReturnPlural<Res, Key>, TOpt>
   : // otherwise access plain key without adding plural and ordinal suffixes
   Res extends readonly unknown[]
   ? Key extends `${infer NKey extends number}`
-    ? Res[NKey]
+    ? ParseTReturnJoinArrays<Res[NKey], TOpt>
     : never
-  : Res[Key & keyof Res];
+  : ParseTReturnJoinArrays<Res[Key & keyof Res], TOpt>;
 
 type TReturnOptionalNull = _ReturnNull extends true ? null : never;
-type TReturnOptionalObjects<TOpt extends TOptions> = _ReturnObjects extends true
+type TReturnOptionalObjects<TOpt extends TOptions> = _JoinArrays extends string
```

Comment:
This should be change to rely on `TOpt` type parameter (review main comment)
```suggestion
type TReturnOptionalObjects<TOpt extends TOptions> = TOpt['joinArrays'] extends string
```